src/lib/coaler/io/FileParser.cpp

```diff
@@ -56,6 +57,26 @@ namespace coaler::io {
             throw std::runtime_error("Unsupported file extension: " + file_extension);
         }
 
-        return result;
+        return checkInputMolecules(result, file_path);
+    }
+
+    RDKit::MOL_SPTR_VECT FileParser::checkInputMolecules(const RDKit::MOL_SPTR_VECT mols, const std::string file_path) {
+        RDKit::MOL_SPTR_VECT retMols;
+        std::vector<std::string> smilesVec;
+        unsigned duplicates = 0;
+        for (auto mol : mols) {
+            std::string smiles = RDKit::MolToSmiles(*mol);
+            if (std::find(smilesVec.begin(), smilesVec.end(), smiles) == smilesVec.end()) {
```

Comment:
since this is ran only once its not a big deal, but id still consider using a hashmap for duplicate removal. should be much more efficient, this does O(n²) string comparisons